internal/engine.go

```diff
@@ -51,7 +73,35 @@ func (e *Engine) initDefaultRules() {
 		&DeferRule{},
 		&MissingModPackageRule{},
 	}
-	e.registerDefaultRules()
+	e.allRules = append(e.allRules, allRules...)
+}
+
+func (e *Engine) registerDefaultRules() {
+	e.defaultRules = []LintRule{
+		NewGolangciLintRule(),
+		NewDeprecateFuncRule(),
+		NewEarlyReturnOpportunityRule(),
+		NewSimplifySliceExprRule(),
+		NewUnnecessaryConversionRule(),
+		NewLoopAllocationRule(),
+		NewEmitFormatRule(),
+		NewDetectCycleRule(),
+		NewGnoSpecificRule(),
+		NewRepeatedRegexCompilationRule(),
+		NewUselessBreakRule(),
+		NewDeferRule(),
+		NewMissingModPackageRule(),
+	}
+	e.rules = append(e.rules, e.defaultRules...)
+}
```

Comment:
I'm not sure about using `registerAllRules` and `registerDefaultRules` functions for initializing rules. I believe there might be a another better approach.

maybe like this:

```go
type ruleConstructor func() LintRule
type ruleMap map[string]ruleConstructor

var allRuleConstructors = ruleMap{
    "golangci-lint": NewGolangciLintRule,
    "deprecate-func": NewDeprecateFuncRule,
    // ...
}

func (e *Engine) registerRules(ruleNames []string) {
    for _, name := range ruleNames {
        if constructor, ok := allRuleConstructors[name]; ok {
            rule := constructor()
            e.rules = append(e.rules, rule)
            e.allRules = append(e.allRules, rule)
        }
    }
}
```

Reply:
went through maps only with this commit https://github.com/gnoverse/tlin/pull/91/commits/ca6a4e47b441e7a4a9107b3b678c179f195956c7, tell me what you think about it